packages/plugin-expo-device/device.js

```diff
@@ -20,10 +22,22 @@ module.exports = {
     // get the initial orientation
     updateOrientation()
 
+    // Fetch the free disk space up front because it's an async API so we don't
+    // want to do this in the onError callback. This means the reported free
+    // disk space could be inaccurate as it may change between now and when an
+    // error occurs, however it's unlikely to be drastically different
+    let freeDisk
+    FileSystem.getFreeDiskStorageAsync().then(freeDiskStorage => { freeDisk = freeDiskStorage })
```

Comment:
This promise has no catch clause so it could cause an unhandled promise rejection.

It needs something like `.catch(er => client._logger.warn('Could not detect free disk space', er)`.